src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,55 @@
+.calendar {
+  width: $calendar-width;
+
+  display: flex;
+  justify-content: flex-start;
+  align-items: flex-end;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+
+  @include content-padding;
+
+  @for $number from 1 through 31 {
+    & :nth-child(#{$number})::before {
+      content: '#{$number}';
+    }
+  }
+
+
+  &__day {
+    height: $day-block-size;
+    width: $day-block-size;
+    box-sizing: border-box;
+    position: relative;
+    display: none;
+
+    background: $day-default-color;
+    border: 1px solid $day-border-color;
+
+    font-size: 30px;
+  }
+
+  @for $count from 1 through $max-month-length {
+    &-month-length--#{$count} :nth-child(-n + #{$count}) {
+      display: flex;
```

Comment:
it's easier to do vice versa (less amount of final CSS code): in `&__day` set normal styles instead of `display` none

and here set display none for children with a number greater than `month-length--${count}` count
note the min number of days in the month is 28, so you don't need create such styles from 1 number